main.cpp

```diff
@@ -299,12 +303,14 @@ struct IntType
     IntType& apply( std::function<IntType&( int& )> func )
     {
         if ( func != nullptr )
-            func( *a );
-        return *this;
+            return func( *a );
+        else
+            std::cout << "Warning, nullptr, can't apply.\n";
+            return *this;
```

Comment:
You didn't use `{ }` so this `return *this` is not actually a part of that `else` expression.
- [ ] fix your indents.